### PR TITLE
[ADD] add module stock_backorder_strategy

### PR DESCRIPTION
**Pull request by laetitia-gangloff**
Define on stock_picking_type how to manage backorder.
create : default behavior
no_create : no backorder created
cancel : backorder is cancelled

### CODE REVIEW

File: stock_backorder_strategy/README.rst [hunk 1]
```diff
@@ -0,0 +1,60 @@
+.. image:: https://img.shields.io/badge/licence-AGPL--3-blue.svg
+    :alt: License: AGPL-3
+
+Stock No Backorder
+==================
+
+In types of operation add a backorder strategy to be able to cancel created
+backorder or to avoid to create a backorder.
```
Comment (pedrobaeza) [inline stock_backorder_strategy/README.rst:8]: Please explain here the use cases for each backorder strategy

File: stock_backorder_strategy/README.rst [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+============
+
+To install this module, you need to:
+
+* Click on install button
+
+Configuration
+=============
+
+To configure this module, you need to:
```
Comment (pedrobaeza) [inline stock_backorder_strategy/README.rst:26]: Put here the menu path for setting this.

File: stock_backorder_strategy/__openerp__.py [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+
+{
+    "name": "Stock no backorder",
```
Comment (pedrobaeza) [inline stock_backorder_strategy/__openerp__.py]: Picking backordering strategies

File: stock_backorder_strategy/__openerp__.py [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+    "version": "0.1",
+    'author': "Acsone, Odoo Community Association (OCA)",
```
Comment (pedrobaeza) [inline stock_backorder_strategy/__openerp__.py]: You usually put "ACSONE A/V" as the author

File: stock_backorder_strategy/models/stock.py [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+            res = True
+        else:
+            res = super(StockPicking, self)._create_backorder(
+                picking, backorder_moves=backorder_moves)
+            if res and picking.picking_type_id.backorder_strategy == 'cancel':
+                self.browse(res).action_cancel()
+        return res
+
+
+class stock_picking_type(models.Model):
```
Comment (pedrobaeza) [inline stock_backorder_strategy/models/stock.py]: CamelCase for class name

File: stock_backorder_strategy/tests/test_backorder_strategy.py [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+
+import openerp.tests.common as common
+
+
+class TestBackorderStrategy(common.TransactionCase):
+
+    def _process_picking(self, picking_type):
```
Comment (pedrobaeza) [inline stock_backorder_strategy/tests/test_backorder_strategy.py]: Why don't make this on setUp and then put each case on a isolated method? As it's rollbacked, you don't need to create 3 pickings, but just change the picking type backordering strategy

**Comment by laetitia-gangloff** [comment]
@pedrobaeza Thank you very much !!

I adapt following your usefull comments.

File: stock_backorder_strategy/README.rst [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+
+* define backorder strategy on picking type (Warehouse > Configuration > Types of Operation)
+
+Usage
+=====
+
+For further information, please visit:
```
Comment (pedrobaeza) [inline stock_backorder_strategy/README.rst:37]: You can also add the new cool button "Try me on runbot":

```
.. image:: https://odoo-community.org/website/image/ir.attachment/5784_f2813bd/datas
   :alt: Try me on Runbot
   :target: https://runbot.odoo-community.org/runbot/154/8.0
```

**Comment by pedrobaeza** [comment]
Thanks for the changes. One last minor remark, but :+1: when CIs goes green

File: stock_backorder_strategy/README.rst [hunk 8]
```diff
@@ -0,0 +1,66 @@
+.. image:: https://img.shields.io/badge/licence-AGPL--3-blue.svg
+    :alt: License: AGPL-3
+
+Stock No Backorder
+==================
+
+In types of operation add a backorder strategy to be able to cancel created
+backorder or to avoid to create a backorder.
+
+Strategies possible are :
```
Comment (yvaucher) [inline stock_backorder_strategy/README.rst]: Would be better to write:

`Possible strategies are:`

**Comment by yvaucher** [comment]
Just one nitpicking otherwise :+1:

**Comment by laetitia-gangloff** [comment]
@yvaucher @pedrobaeza thank you :smile:

**Comment by laetitia-gangloff** [comment]
Should I change the repository for stock-logistics-workflow (instead of stock-logistics-warehouse) ?

**Comment by pedrobaeza** [comment]
Uhm, good question. Yeah, I think so, because pickings are transactional data that corresponds more to the workflow. A renaming to _stock_picking_backorder_strategy_ also seems reasonable.

**Comment by laetitia-gangloff** [comment]
sorry for the mistake, I migrate this PR here : https://github.com/OCA/stock-logistics-workflow/pull/143
I close this one.

**Pull request closed without merging**